vignettes/epipredict.Rmd

```diff
@@ -1,470 +1,565 @@
+          ahead = 14
+        )
+      )
+  })
+map_df(all_fits, ~ pluck(., "predictions"))
 ```
 
-Here we have 5 layers of frosting. The first generates the forecasts from the test data.
-The second uses quantiles of the residuals to create distributional
-forecasts. The next two add columns for the date the forecast was made and the
-date for which it is intended to occur. Because we are predicting rates, they
-should be non-negative, so the last layer thresholds both predicted values and
-intervals at 0. The code to do this (inside the forecaster) is
-
-```{r}
-f <- frosting() %>%
-  layer_predict() %>%
-  layer_residual_quantiles(
-    quantile_levels = c(.01, .025, 1:19 / 20, .975, .99),
-    symmetrize = TRUE
-  ) %>%
-  layer_add_forecast_date() %>%
-  layer_add_target_date() %>%
-  layer_threshold(starts_with(".pred"))
+Fitting separate models for each geography is both 56 times slower[^7] than geo-pooling, and fits each model on far less data.
+If a dataset contains relatively few observations for each geography, fitting a geo-pooled model is likely to produce better, more stable results.
+However, geo-pooling can only be used if values are comparable in meaning and scale across geographies or can be made comparable, for example by normalization.
+
+If we wanted to build a geo-aware model, such as a linear regression with a different intercept for each geography, we would need to build a [custom workflow](custom_epiworkflows) with geography as a factor.
+
+# Anatomy of a canned forecaster
+## Code object
+Let's dissect the forecaster we trained back on the [landing
+page](../index.html#motivating-example):
+
+```{r make-four-forecasts, warning=FALSE}
+four_week_ahead <- arx_forecaster(
+  covid_case_death_rates |> filter(time_value <= forecast_date),
+  outcome = "death_rate",
+  predictors = c("case_rate", "death_rate"),
+  args_list = arx_args_list(
+    lags = list(c(0, 1, 2, 3, 7, 14), c(0, 7, 14)),
+    ahead = 4 * 7,
+    quantile_levels = c(0.1, 0.25, 0.5, 0.75, 0.9)
+  )
+)
 ```
 
-At predict time, we add this object onto the `epi_workflow` and call `forecast()`
+`four_week_ahead` has two components: an `epi_workflow`, and a table of
+`predictions`.
+The table of predictions is a simple tibble,
 
-```{r, warning=FALSE}
-ewf %>%
-  add_frosting(f) %>%
-  forecast()
+```{r show_predictions}
+four_week_ahead$predictions
 ```
 
-The above `get_test_data()` function examines the recipe and ensures that enough
-test data is available to create the necessary lags and produce a prediction
-for the desired future time point (after the end of the training data). This mimics
-what would happen if `jhu` contained the most recent available historical data and
-we wanted to actually predict the future. We could have instead used any test data
-that contained the necessary predictors.
+where `.pred` gives the point/median prediction, and `.pred_distn` is a
+`dist_quantiles()` object representing a distribution through various quantile
```

Comment:
It no longer has this class: it is `quantile_pred`. The rest of this paragraph
is similarly out of date.